File: backend/app/schemas/publish.py
from datetime import datetime

from pydantic import BaseModel


class PublishRequest(BaseModel):
    thumb_media_id: str = ""


class PublishRecordOut(BaseModel):
    id: int
    content_id: int
    channel: str
    status: str
    platform_article_id: str | None
    platform_url: str | None
    published_at: datetime | None
    error_message: str | None
    created_at: datetime
    community_id: int

    model_config = {"from_attributes": True}


class PublishRecordListOut(BaseModel):
    total: int
    items: list[PublishRecordOut]


class ChannelPreview(BaseModel):
    channel: str
    title: str
    content: str
    format: str  # "html" or "markdown"


class CopyContent(BaseModel):
    channel: str
    title: str
    content: str
    format: str
    platform: str


class AnalyticsOut(BaseModel):
    id: int
    publish_record_id: int
    read_count: int
    like_count: int
    share_count: int
    comment_count: int
    collected_at: datetime

    model_config = {"from_attributes": True}


class AnalyticsOverview(BaseModel):
    total_contents: int
    total_published: int
    channels: dict[str, int]  # channel -> published count


class ContentAnalyticsDetail(BaseModel):
    """内容分析详情（含发布记录列表）"""
    content_id: int
    title: str
    analytics: list["PublishRecordOut"] = []

    model_config = {"from_attributes": True}


class ChannelConfigOut(BaseModel):
    id: int
    channel: str
    config: dict
    enabled: bool

    model_config = {"from_attributes": True}


class ChannelConfigCreate(BaseModel):
    channel: str  # wechat, hugo, csdn, zhihu, ...
    config: dict = {}
    enabled: bool = False


class ChannelConfigUpdate(BaseModel):
    config: dict = {}
    enabled: bool | None = None

File: frontend/src/api/publish.ts
import apiClient from './index'

const api = apiClient

export interface PublishRecord {
  id: number
  content_id: number
  channel: string
  status: string
  platform_article_id: string | null
  platform_url: string | null
  published_at: string | null
  error_message: string | null
  created_at: string
}

export interface ChannelPreview {
  channel: string
  title: string
  content: string
  format: string
}

export interface CopyContent {
  channel: string
  title: string
  content: string
  format: string
  platform: string
}

export interface AnalyticsOverview {
  total_contents: number
  total_published: number
  channels: Record<string, number>
}

export interface ChannelConfig {
  id: number
  channel: string
  config: Record<string, string>
  enabled: boolean
}

export async function publishToWechat(contentId: number): Promise<PublishRecord> {
  const { data } = await api.post(`/publish/${contentId}/wechat`)
  return data
}

export async function publishToHugo(contentId: number): Promise<PublishRecord> {
  const { data } = await api.post(`/publish/${contentId}/hugo`)
  return data
}

export async function getPreview(contentId: number, channel: string): Promise<ChannelPreview> {
  const { data } = await api.get(`/publish/${contentId}/preview/${channel}`)
  return data
}

export async function getCopyContent(contentId: number, channel: string): Promise<CopyContent> {
  const { data } = await api.get(`/publish/${contentId}/copy/${channel}`)
  return data
}

export async function getPublishRecords(contentId?: number): Promise<PublishRecord[]> {
  const { data } = await api.get('/publish/records', { params: contentId ? { content_id: contentId } : {} })
  return data
}

export async function getAnalyticsOverview(): Promise<AnalyticsOverview> {
  const { data } = await api.get('/analytics/overview')
  return data
}

export async function getPublishTrend(days = 30): Promise<{ items: { date: string; count: number }[]; days: number }> {
  const { data } = await api.get('/analytics/trend/daily', { params: { days } })
  return data
}

export async function getChannelConfigs(): Promise<ChannelConfig[]> {
  const { data } = await api.get('/analytics/settings/channels')
  return data
}

export async function updateChannelConfig(
  channel: string,
  payload: { config?: Record<string, string>; enabled?: boolean }
): Promise<ChannelConfig> {
  const { data } = await api.put(`/analytics/settings/channels/${channel}`, payload)
  return data
}
